File: beecrowd/beginners/1018.py
'''
1018 - CÉDULAS

Leia um valor inteiro. A seguir, calcule o menor número de notas possíveis (cédulas) no qual o valor pode ser decomposto. As notas consideradas são de 100, 50, 20, 10, 5, 2 e 1. A seguir mostre o valor lido e a relação de notas necessárias.

Entrada
O arquivo de entrada contém um valor inteiro N (0 < N < 1000000).

Saída
Imprima o valor lido e, em seguida, a quantidade mínima de notas de cada tipo necessárias, conforme o exemplo fornecido. Não esqueça de imprimir o fim de linha após cada linha, caso contrário seu programa apresentará a mensagem: “Presentation Error”.
'''
valor = int(input())

print(valor)
notas = valor // 100
print(f"{notas} nota(s) de R$ 100,00")
valor = valor % 100

notas = valor // 50
print(f"{notas} nota(s) de R$ 50,00")
valor = valor % 50

notas = valor // 20
print(f"{notas} nota(s) de R$ 20,00")
valor = valor % 20

notas = valor // 10
print(f"{notas} nota(s) de R$ 10,00")
valor = valor % 10

notas = valor // 5
print(f"{notas} nota(s) de R$ 5,00")
valor = valor % 5

notas = valor // 2
print(f"{notas} nota(s) de R$ 2,00")
valor = valor % 2

notas = valor // 1
print(f"{notas} nota(s) de R$ 1,00")
valor = valor % 1

File: beecrowd/cpp/beginners/1018.cpp
#include <iostream>

int main() {
    int n, valor, resto;
    std::cin >> n;
    std::cout << n << std::endl;

    valor = n / 100;
    resto = n % 100;
    std::cout << valor << " nota(s) de R$ 100,00" << std::endl;

    valor = resto / 50;
    resto = resto % 50;
    std::cout << valor << " nota(s) de R$ 50,00" << std::endl;

    valor = resto / 20;
    resto = resto % 20;
    std::cout << valor << " nota(s) de R$ 20,00" << std::endl;

    valor = resto / 10;
    resto = resto % 10;
    std::cout << valor << " nota(s) de R$ 10,00" << std::endl;

    valor = resto / 5;
    resto = resto % 5;
    std::cout << valor << " nota(s) de R$ 5,00" << std::endl;

    valor = resto / 2;
    resto = resto % 2;
    std::cout << valor << " nota(s) de R$ 2,00" << std::endl;

    valor = resto / 1;
    resto = resto % 1;
    std::cout << valor << " nota(s) de R$ 1,00" << std::endl;
}